### ISSUE TITLE
Packaging nteract application

### ISSUE THREAD

**Issue by captainsafia** [nteract/archived-desktop-app#548]
Transferring this over from the Sprint Structure hackpad.
- [ ] Associator work (registering filetypes to be opened by nteract)
- [ ] 0mq prebuilds

cc: @rgbkrk @jdfreder

**Comment by rgbkrk** [comment]
I just packaged a release and am running into this issue:

```
Uncaught Exception:
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at basename (path.js:1355:5)
    at Command.parse (/Users/kyle/code/src/github.com/nteract/nteract/dist/nteract-darwin-x64/nteract.app/Contents/Resources/app/node_modules/commander/index.js:446:30)
    at Object.<anonymous> (/Users/kyle/code/src/github.com/nteract/nteract/dist/nteract-darwin-x64/nteract.app/Contents/Resources/app/build/main/index.js:14:64)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Module.require (module.js:466:17)
```

**Comment by rgbkrk** [comment]
Same thing if I run the nteract app with a file. Digging into it now.

**Comment by captainsafia** [comment]
@jdetle and I ran into this yesterday when we were pairing on his Linux machine, as well.

**Comment by jdetle** [comment]
@rgbkrk I am trying to work through this too :+1:

**Comment by rgbkrk** [comment]
It's only happening for me on the version built with our `npm run dist:osx` setup. I'm beginning to think this is an issue of the electron version we're on right now, based on my debugging. Since we need to get up to date anyways, I'm going to try updating electron.

**Comment by jdetle** [comment]
Okay I figured it out!

**Comment by jdetle** [comment]
Commander.parse doesn't like undefined as an argument. If you go ahead and hack your index.js code in `src/main` 

```
const program = require('commander');
const version = require('../../package.json').version;
process.argv[1] = ''
program
  .version(version)
  .option('-k', '--kernel [kernel]', '')
  .parse(process.argv);
```

it will run. I imagine newer versions of commander have fixed this problem. 
We can go ahead an iterate over argv to check for undefined and replace it with the empty string as a hacky fix?

**Comment by jdetle** [comment]
It also might be worth considering moving to vorpal? https://github.com/dthree/vorpal/

Update: after playing around with vorpal, I give up!

**Comment by jdetle** [comment]
@rgbkrk @captainsafia  if you run ./nteract 'any string here', I think it will open an untitled notebook. If you run ./nteract 'any string here' 'name of file in your path here' it will open your file. 

Another current error is that if the specified file doesn't exist in the path it will hang and nothing will open.

**Comment by jdetle** [comment]
Pair programmed with @captainsafia and we came to the conclusion that commander eats the first argument in an attempt to resolve the name of the node app that is being ran. 

Under the tutelage of @captainsafia, I tried to play around with process.argv to extract the notebook info ourselves, but in reality we should probably go ahead and understand a tool like commander or vorpal better to get the appropriate behavior. 

I'll come back tomorrow with fresh eyes in an attempt to have an elegant fix.

**Comment by captainsafia** [comment]
> Under the tutelage of @captainsafia

Harhar! More like under the trouble-making.

`commander` is a little weird in some cases and the docs are tough to grok without actually looking into the source. Have you had any luck, @rgbkrk?

**Comment by jdetle** [comment]
Ahah! I fixed the problem by editing one line of code in `node_modules/commander/index.js` changing line 456 to 
  `var parsed = this.parseOptions(this.normalize(argv.slice(1)));` from `
  var parsed = this.parseOptions(this.normalize(argv.slice(2)));` makes everything behave swell.

I know this isn't how we want to go about fixing the problem, but it helps us understand whats going on. Edit: I neglected to mention that I added program._name...

```
program._name = 'nteract';

program
  .version(version)
  .option('-k, --kernel [kernel]', 'kernel')
  .parse(process.argv); // this is the edited code changing the 2 to a 1
```
